jupyterhub/templates/hub/configmap.yaml

```diff
@@ -5,312 +5,18 @@ metadata:
-  kubespawner.common-labels: |
-    {{- $_ := merge (dict "heritageLabel" "jupyterhub") . }}
-    {{- include "jupyterhub.commonLabels" $_ | nindent 4 }}
-
-
-  {{- /* Hub */}}
-  hub.allow-named-servers: {{ .Values.hub.allowNamedServers | quote }}
-  hub.concurrent-spawn-limit: {{ .Values.hub.concurrentSpawnLimit | quote }}
-  hub.consecutive-failure-limit: {{ .Values.hub.consecutiveFailureLimit | quote }}
-  {{- if .Values.hub.activeServerLimit }}
-  hub.active-server-limit: {{ .Values.hub.activeServerLimit | quote }}
-  {{- end }}
-  {{- if .Values.hub.services }}
-  hub.services: |
-    {{- /* include hub.services *except* api_tokens */}}
-    {{- range $name, $service := .Values.hub.services }}
-    {{ $name }}:
-      {{- range $key, $value := $service }}
-      {{- if ne $key "apiToken" }}
-      {{- /* Use toJson since values can be more than just strings */}}
-      {{ $key }}: {{ toJson $value }}
-      {{- end }}
-      {{- end }}
-    {{- end }}
-  {{- end }}
-  {{- if .Values.hub.extraConfig }}
-  {{- $extraConfigType := typeOf .Values.hub.extraConfig }}
-  {{- if eq $extraConfigType "string" }}
-  hub.extra-config.default.py: |
-    {{- .Values.hub.extraConfig | nindent 4 }}
-  {{- else if eq $extraConfigType "map[string]interface {}" }}
-  {{- range $key, $value := .Values.hub.extraConfig }}
-  hub.extra-config.{{ $key }}.py: |
-    {{- $value | nindent 4 }}
-  {{- end }}
-  {{- end }}
-  {{- end }}
-  {{- if .Values.hub.extraConfigVars }}
-  hub.extra-vars: {{ toJson .Values.hub.extraConfigVars | quote }}
-  {{- end }}
-  hub.base_url: {{ .Values.hub.baseUrl | quote }}
-  {{- if eq .Values.hub.db.type "sqlite-pvc" }}
-  hub.db_url: "sqlite:///jupyterhub.sqlite"
-  {{- else if eq .Values.hub.db.type "sqlite-memory" }}
-  hub.db_url: "sqlite://"
-  {{- else }}
-  hub.db_url: {{ .Values.hub.db.url | quote }}
-  {{- end }}
-  {{- if .Values.hub.cookieSecret }}
-  token.cookie_secret: {{ .Values.hub.cookieSecret | quote }}
-  {{- end }}
-
-  {{- /* Other */}}
-  debug.enabled: {{ .Values.debug.enabled | quote }}
-  {{- range $key, $value := .Values.hub.extraConfigMap }}
-  custom.{{ $key }}: {{ toJson $value | quote }}
-  {{- end }}
+{{- $values := merge dict .Values -}}
+{{- /* trim secret values. Update here if new secrets are added! */ -}}
+{{- $_ := set $values "proxy" (omit $values.proxy "secretToken" "https.manual") -}}
+{{- $_ := set $values.proxy "https" (omit $values.proxy.https "manual") -}}
+{{- $_ := set $values "hub" (omit $values.hub "cookieSecret") -}}
+{{- /* make a copy of values.auth to avoid modifying the original */ -}}
+{{- $_ := set $values "auth" (merge dict .Values.auth) -}}
+{{- $_ := set $values.auth "state" (omit $values.auth.state "cryptoKey") -}}
+{{- range $key, $service := $values.hub.services -}}
+  {{- if $service.apiToken -}}
+    {{- $_ := set $values.hub.services $key (omit $service "apiToken") -}}
+  {{- end -}}
+{{- end }}
+  values.yaml: |
+    {{- $values | toYaml | nindent 4 }}
```

Comment:
We are always using  `| trimSuffix "\n"` in the current templates after `toYaml` to avoid an extra new line, we could relax the linter here and stop doing so, but this isn't only a linting issue, but mostly is.

The new line generated, and the indentation from the `nindent` following it, is generated content, any whitespace chomping left you attempt to do will make you end up with that indentation. I recall that has been an issue, that is why I was frustrated enough to add the ugly practice of `trimSuffix "\n"` after `toYaml` statements. I'm not confident this is still an issue.

For this PR TODO:
- Pipe `toYaml` into `trimSuffix "\n"`

For us TODO in future:
- See if we can clean up our code from needing to use `trimSuffix "\n"` by ignoring the linter without getting into trouble with the actual validity of generated yaml.

Reply:
Since the linter doesn't complain, I don't think there's anything to do here. If the linter did complain, it's the linter that's the problem, but since this is both the more desirable code and producing no lint errors, there's nothing to do, right?